django/db/models/sql/compiler.py

```diff
@@ -631,6 +633,22 @@ def _get_field_choices():
             )
             return chain(direct_choices, reverse_choices)
 
+        def _build_error_msg(field_name):
+            try:
+                field = opts.get_field(field_name)
+            except FieldDoesNotExist:
+                return "'%s' (reason: non existing field)" % field_name
+            if field_name in opts.fields_map:
+                return "'%s' (reason: reverse relational field)" % field_name
+            if field.is_relation and field.many_to_many:
+                return "'%s' (reason: many to many field)" % field_name
+            if field.is_relation and field.one_to_many:
+                return "'%s' (reason: generic relation)" % field_name
+            if (field.is_relation and field.many_to_one and
```

Comment:
kind of a duplicate of nested function is_not_a_generic_foreign_key in django.db.models.options. Maybe we should consider making this function a public API?